modules/product/src/main/java/com/opengamma/strata/product/rate/BrlCompoundedOvernightRateComputation.java

```diff
@@ -0,0 +1,545 @@
+/*
+ * Copyright (C) 2018 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.rate;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.joda.beans.Bean;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.ImmutableValidator;
+import org.joda.beans.gen.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.date.HolidayCalendar;
+import com.opengamma.strata.basics.index.OvernightIndex;
+import com.opengamma.strata.collect.ArgChecker;
+
+/**
+ * Defines the computation of a rate from that follows the compounding of a floating BRL swap leg.
+ * <p>
+ * An interest rate determined directly from an Overnight index that follows the compounding of a floating BRL swap leg.
+ */
+@BeanDefinition
+public final class BrlCompoundedOvernightRateComputation
+    implements OvernightRateComputation, ImmutableBean, Serializable {
+
+  /**
+   * The Overnight index.
+   * <p>
+   * The rate to be paid is based on this index.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final OvernightIndex index;
+
```

Comment:
General Strata rule is no blank lines between fields (as they have multi-line javadoc)